python/tritonserver/_api/wrapper.py

```diff
@@ -0,0 +1,725 @@
+
+        return options
+
+
+class ModelDictionary(dict):
+    def __init__(self, server, models):
+        super().__init__()
+        for model in models:
+            self[(model.name, model.version)] = model
+        self._server = server
+        self._model_names = [x[0] for x in self.keys()]
+
+    def __getitem__(self, key):
+        if isinstance(key, tuple):
+            try:
+                return dict.__getitem__(self, key)
+            except KeyError:
+                raise KeyError(f"Unknown Model: {key}") from None
+        else:
+            if key in self._model_names:
+                return Model(self._server, name=key, version=-1)
+            else:
+                raise KeyError(f"Unknown Model: {key}")
+
+
+class Server:
+    def __init__(self, options: Options = None, **kwargs):
+        if options is None:
+            options = Options(**kwargs)
+        self._options = options
+        self._server = Server._UnstartedServer()
+
+    def start(self, blocking=False, polling_interval=0.1):
+        if not isinstance(self._server, Server._UnstartedServer):
+            raise triton_bindings.InvalidArgumentError("Server already started")
+
+        self._server = triton_bindings.TRITONSERVER_Server(
+            self._options._create_server_options()
+        )
+        while blocking and not self.is_ready():
+            time.sleep(polling_interval)
+
+    def stop(self):
+        self._server.stop()
+        self._server = Server.UnstartedServer()
+
+    def unregister_model_repository(self, repository_path: str):
+        self._server.unregister_model_repository(repository_path)
+
+    def register_model_repository(
+        self, repository_path: str, name_mapping: dict[str, str]
+    ):
+        """Add a new model repository.
+
+        Adds a new model repository.
+
+        See :c:func:`TRITONSERVER_ServerRegisterModelRepository`
+
+        Parameters
+        ----------
+        repository_path : str
+            repository path
+        name_mapping : dict[str, str]
+            override model names
+
+        Examples
+        --------
+        server.register_model_repository("/workspace/models",{"test_model":"new_model"})
+
+        """
+        name_mapping_list = [
+            triton_bindings.TRITONSERVER_Parameter(name, value)
+            for name, value in name_mapping.items()
+        ]
+
+        self._server.register_model_repository(repository_path, name_mapping_list)
+
+    def poll_model_repository(self):
+        return self._server.poll_model_repository()
+
+    def metadata(self):
+        return json.loads(self._server.metadata().serialize_to_json())
+
+    def is_live(self):
+        return self._server.is_live()
+
+    def is_ready(self):
+        return self._server.is_ready()
+
+    def get_model(self, model_name, model_version=-1):
+        return Model(self._server, model_name, model_version)
+
+    @property
+    def models(self):
```

Comment:
is there a way to access a specific model version in this API or the user has to use `get_model`?

Reply:
models[("test",1)] 

You can give either a string or a tuple.

I also considered models["test"][1] - but the mechanics became a little convoluted and I felt "string or tuple" was fine enough